client/js/screens/UserProfile/UserProfileContainer.js

```diff
@@ -8,21 +8,37 @@ import FavesContext from '../../context/FavesContext';
 import CircularLoader from '../../components/CircularLoader';
 import {withNavigation} from 'react-navigation';
 import {USER_QUERY, ALL_INTERESTS} from '../../config/apollo/queries';
+import Modal from 'react-native-modal';
 
 class UserProfileContainer extends Component {
+  state = {
+    isModalVisible: false,
+  };
+
+  toggleModal = () => {
+    this.setState({isModalVisible: !this.state.isModalVisible});
+  };
+  componentDidMount = () => {
+    this.props.navigation.setParams({toggleModal: this.toggleModal});
+  };
+
+  logOut = async () => {
+    try {
+      removeToken();
+      this.props.navigation.navigate('AuthLoading');
+    } catch (e) {
+      throw Error(e);
+    }
+  };
   static navigationOptions = ({navigation, screenprops}) => ({
     title: 'Profile',
     headerRight: (
-      <TouchableOpacity
-        onPress={async () => {
-          try {
-            removeToken();
-            navigation.navigate('AuthLoading');
-          } catch (e) {
-            throw Error(e);
-          }
-        }}>
-        <Text style={styles.logout}>Sign Out </Text>
+      <TouchableOpacity onPress={navigation.getParam('toggleModal')}>
+        {navigation.getParam('user') ? (
+          <Text></Text>
```

Comment:
instead of setting empty `<Text/>`, would it work if you change to

```
headerRight: navigation.getParam('user') ? (
  <TouchableOpacity onPress={navigation.getParam('toggleModal')}>
    <Text style={styles.logout}>Sign Out</Text>
  </TouchableOpacity>
) : null,
```

in this case the `Touchable` component is only created when you can sign out